### PR TITLE
SVCPLAN-785 - create profile_lvm

### PR DESCRIPTION
**Pull request by jakerundall**
define profile_lvm, using lvm module

include support for bindmounts from (subdirectories within) LVs

### CODE REVIEW

File: manifests/bindmounts.pp [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+#       src_path: "/slurm/slurmctld.state"
+# ```
+#
+# @example
+#   include profile_lvm::bindmounts
+class profile_lvm::bindmounts (
+  Optional[ Hash ] $map = undef,
+) {
+
+  $map.each | $k, $v | {
```
Comment (billglick) [inline manifests/bindmounts.pp]: In the `lvm` module port, we wrapped essentially this same logic with an `if $map`. Do we need that same logic here?

File: manifests/init.pp [hunk 2]
```diff
@@ -44,9 +44,9 @@
 #   
 class profile_lvm (
 
-  String                $default_fs_type,
-  Hash[String[1],Hash ] $lvs,
-  Array[String]         $required_pkgs,
+  Optional[String]                $default_fs_type,
+  Optional[Hash[String[1],Hash ]] $lvs,
+  Optional[Array]                 $required_pkgs,
```
Comment (billglick) [inline manifests/init.pp]: Why are these optional?

We actually provide all of them via the module's default data.

File: README.md [hunk 3]
```diff
@@ -32,7 +35,23 @@ include ::profile_lvm
...
+```
+
+Optionally define non-standard required packages or a custom default
+filesystem type.
+
+Optionally define bindmounts from LVs (or sub-directories contained in
+LVs).
+
 
 ## Dependencies
```
Comment (billglick) [inline README.md:72]: Link in the ncsa/lvm ported module

File: README.md [hunk 4]
```diff
@@ -32,7 +35,23 @@ include ::profile_lvm
...
+    size: "100G"
+    size_is_minsize: true
+    volume_group: "data"
+```
+
+Optionally define non-standard required packages or a custom default
+filesystem type.
+
+Optionally define bindmounts from LVs (or sub-directories contained in
+LVs).
```
Comment (billglick) [inline README.md]: Can you add example hiera for this? Ideally related to the `lvs` example above.

**Review by billglick — CHANGES_REQUESTED** [review]
A few minor suggestions and questions...

File: manifests/bindmounts.pp (hunk 1 above)
Comment (jakerundall) [reply manifests/bindmounts.pp]: Yes! Updating.

File: manifests/init.pp (hunk 2 above)
Comment (jakerundall) [reply manifests/init.pp]: Fixed.

File: README.md (hunk 3 above)
Comment (jakerundall) [reply README.md:72]: Done.

File: README.md (hunk 4 above)
Comment (jakerundall) [reply README.md]: Added that example (for slurm, and switched the original lv to slurm as well) and examples for a couple of the other params.

**Comment by jakerundall** [comment]
@billglick Should be all fixed and ready for you to take another look. Thanks!

**Comment by jakerundall** [comment]
NOTE: Somehow the main branch got created as master, so after I merge I will be renaming that to main.

**Review by billglick — APPROVED** [review]
Looks good.

**Pull request merged**